core/src/main/scala/cats/collections/Predicate.scala

```diff
@@ -63,12 +63,20 @@ object Predicate extends PredicateInstances {
 }
 
 trait PredicateInstances {
+  implicit def predicateContravariantMonoidal: ContravariantMonoidal[Predicate] = new ContravariantMonoidal[Predicate] {
+    override def contramap[A, B](fb: Predicate[A])(f: B => A): Predicate[B] =
+      Predicate(f andThen fb.apply)
+    override def product[A, B](fa: Predicate[A], fb: Predicate[B]): Predicate[(A, B)] =
+      Predicate(v => fa(v._1) || fb(v._2))
```

Comment:
I'm curious about the choice of `||` here, what's the reasoning?

Reply:
Yeah, it's definitely a bit arbitrary, but I made the choice based on having the monoidal product act similarly to the monoids already defined for this type. `||` gives us union, which is probably the more in line with user expectations given the monoids are also union.

Reply:
Could you please add law tests for `ContravariantMonoidal`? Then I think we're good to go.

Reply:
Sure, I'll get to that today / tomorrow probably.

Reply:
I did it with kind of stupid `Eq` instances, but I'm not sure there are better ones. If there's a standard way to do `Eq` for functions, let me know and I can change it up.